File: scala/allergies/src/main/scala/allergies.scala
class Allergies() {
  import Allergen.Allergen

  def allergies(score: BigInt): List[Allergen] = {
    Allergen.values.filter(isAllergicTo(_, score)).toList
  }

  def isAllergicTo(allergen: Allergen.Allergen, score: BigInt): Boolean = {
    score testBit allergen.id
  }

}

object Allergies {
  def apply() = new Allergies()
}

object Allergen extends Enumeration {
  type Allergen = Value

  val Eggs = Value(0)
  val Peanuts = Value(1)
  val Shellfish = Value(2)
  val Strawberries = Value(3)
  val Tomatoes = Value(4)
  val Chocolate = Value(5)
  val Pollen = Value(6)
  val Cats = Value(7)
}

File: python/allergies/allergies.py
ITEMS = {
    'eggs':  1,
    'peanuts': 2,
    'shellfish': 4,
    'strawberries': 8,
    'tomatoes': 16,
    'chocolate': 32,
    'pollen': 64,
    'cats': 128
}

class Allergies(object):
  def __init__(self, allergens):
    self._allergens = allergens
    self.lst = [item
                for item in ITEMS.keys()
                if self.is_allergic_to(item)]

  def is_allergic_to(self, item):
    return self._allergens & ITEMS[item] > 0
